src/sentry/integrations/github/repository.py

```diff
@@ -60,7 +62,14 @@ def build_repository_config(
     def compare_commits(
         self, repo: Repository, start_sha: str | None, end_sha: str
     ) -> Sequence[Mapping[str, Any]]:
-        def eval_commits(client: Any) -> Sequence[Mapping[str, Any]]:
+        integration_id = repo.integration_id
+        if integration_id is None:
+            raise NotImplementedError("GitHub apps requires an integration id to fetch commits")
+        integration = Integration.objects.get(id=integration_id)
+        installation: GitHubIntegration = integration.get_installation(repo.organization_id)
```

Comment:
We might need to use the integration_service here instead of using the ORM. Commit comparisons happen in region silos.